src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsProvider.cs

```diff
@@ -254,18 +254,24 @@ protected override void Initialize()
 
                 var launchSettingData = await GetLaunchSettingsAsync().ConfigureAwait(false);
 
-
                 // If there are no profiles, we will add a default profile to run the prroject. W/o it our debugger
                 // won't be called on F5 and the user will see a poor error message
                 if (launchSettingData.Profiles.Count == 0)
                 {
                     launchSettingData.Profiles.Add(new LaunchProfileData() { Name = Path.GetFileNameWithoutExtension(CommonProjectServices.Project.FullPath), CommandName = RunProjectCommandName });
```

Comment:
>launchSettingData.Profiles.Add(new LaunchProfileData() { Name = Path.GetFileNameWithoutExtension(CommonProjectServices.Project.FullPath), CommandName = RunProjectCommandName }); [](start = 20, length = 177)

we left it on purpose with DoNotPersist=false right?  To keep creating profiles on disk when they don't exist?

Reply:
Yes, I debated this one. It probably should be dotnet persist since as soon as a user changes one of its settings it will be persisted. However, if someone adds a new profile, this one won't be saved like it is today, so I purposely left it as is.